utils/docker/images/install-libpmemobj-cpp.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+#
+# Copyright 2019, Intel Corporation
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#     * Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#
+#     * Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in
+#       the documentation and/or other materials provided with the
+#       distribution.
+#
+#     * Neither the name of the copyright holder nor the names of its
+#       contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#
+# install-libpmemobj-cpp.sh <package_type>
+#           - installs PMDK C++ bindings (stable) packages
+#
+
+set -e
+
+git clone https://github.com/pmem/libpmemobj-cpp
+cd libpmemobj-cpp
+# stable-1.5; January 14th
+git checkout b3614479bc5c4b10d138004fa8ff104dd46d5722
```

Comment:
I totally understand why we'd want to specify the dependency like this for a production release -- and this will work until the concurrent map gets merged into libpmemobj-cpp.  But once that map is available we'll have to be building against the libpmemobj-cpp master branch (until its stable-1.6 release is cut).  The INSTRUCTIONS.md file shows building against the master branch version.  I guess this is ok for now, but just be advised that pmemkv will (hopefully in the next few weeks) require pmemobj 1.6 or higher.

Reply:
So, image is only rebuild if one of the files in docker/images is modified - this is performance optimization because rebuild of a docker image takes ages (~10min). So if you want to update dependencies you have to edit this file.

Reply:
FYI, using branch names (master or stable-1.5, doesn't matter) in Docker image build script is not a great idea, because those branches are moving targets - every time build scripts will be updated (typo fix? new example that depends on external library?) it will replace your baseline. If something will break after PR rebase/merge you won't know if it comes from your code or your dependencies.